applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationAsyncClient.java

```diff
@@ -371,7 +510,7 @@ public static ConfigurationAsyncClientBuilder builder() {
      * @throws HttpResponseException If {@code key} is an empty string.
      */
     public Mono<Response<ConfigurationSetting>> deleteSetting(String key) {
-        return deleteSetting(new ConfigurationSetting().key(key));
+        return deleteSetting(new ConfigurationSetting().key(key), Context.NONE);
```

Comment:
This shouldn't be an issue since adding to a Context creates another context using the key-value that was passed and sets the context used for the call to the parent of the new context. Using Context.NONE all over the place just means that there will be a lot of Context linked lists with Context.NONE as the root parent.